src/webgpu/util/texture/base.ts

```diff
@@ -15,8 +15,11 @@ export function maxMipLevelCount({
 }): number {
   const sizeDict = reifyExtent3D(size);
 
-  let maxMippedDimension = sizeDict.width;
-  if (dimension !== '1d') maxMippedDimension = Math.max(maxMippedDimension, sizeDict.height);
+  if (dimension === '1d') {
```

Comment:
WDYT about about doing a switch/case for the dims, and then setting maxMippedDimension in those? (For 1d, I think if we set maxMippedDimension = 1, then the result will be 1 when running the final return statement we already have.)